static/js/reducers/email.js

```diff
@@ -15,38 +16,57 @@ import {
 } from '../actions';
 import type { Action } from '../flow/reduxTypes';
 import type {
-  Email,
-  EmailState
+  AllEmailsState,
+  EmailState,
+  EmailInputs,
 } from '../flow/emailTypes';
 
 export const INITIAL_EMAIL_STATE: EmailState = {
-  email:  {},
+  inputs: {},
   validationErrors: {},
   sendError: {},
 };
+export const INITIAL_ALL_EMAILS_STATE: AllEmailsState = {
+  searchResultEmail: INITIAL_EMAIL_STATE,
+  courseTeamEmail: INITIAL_EMAIL_STATE
+};
 
-export const NEW_EMAIL_EDIT: Email = {
+export const NEW_EMAIL_EDIT: EmailInputs = {
   subject:    null,
   body:       null,
-  query:      null,
 };
 
-export const email = (state: EmailState = INITIAL_EMAIL_STATE, action: Action) => {
+function updatedState(state: AllEmailsState, emailType: string, updateObject: Object) {
+  _.merge(state[emailType], updateObject);
+  return _.cloneDeep(state);
```

Comment:
Just tried it out, and I see why you're not doing it! It seems like `_.merge` does a deep merge (which we want) but it mutates the first argument (which is yucky).

I think we generally do not want to mutate the previous state - it might be better if we do the `_.cloneDeep` of the previous state first, so something like:

```js
return _.merge(_.cloneDeep(state), {[emailType]: updateObject});
```